src/Compilers/CSharp/Portable/Compiler/DocumentationCommentCompiler.cs

```diff
@@ -511,125 +523,211 @@ private bool TryProcessRecordPropertyDocumentation(
+                    _compilation, ref documentedParameters, ref documentedTypeParameters, ref _includedFileCache, expanderWriter, _diagnostics, _cancellationToken);
+            }
+            else if (_writer != null && !shouldSkipPartialDefinitionComments)
             {
-                return TryProcessRecordPropertyDocumentation(recordProperty, docCommentNodes, out withUnprocessedIncludes, out includeElementNodes);
+                // CONSIDER: The output would look a little different if we ran the XDocument through an XmlWriter.  In particular, 
+                // formatting inside tags (e.g. <__tag___attr__=__"value"__>) would be normalized.  Whitespace in elements would
+                // (or should) not be affected.  If we decide that this difference matters, we can run the XDocument through an XmlWriter.
+                // Otherwise, just writing out the string saves a bunch of processing and does a better job of preserving whitespace.
+                Write(withUnprocessedIncludes);
             }
 
-            // We're doing substitution and formatting per-trivia, rather than per-symbol,
-            // because a single symbol can have both single-line and multi-line style
-            // doc comments.
-            foreach (DocumentationCommentTriviaSyntax trivia in docCommentNodes)
+            bool reportParameterOrTypeParameterDiagnostics = GetLocationInTreeReportingDocumentationCommentDiagnostics(symbol) != null;
+            if (reportParameterOrTypeParameterDiagnostics)
             {
                 _cancellationToken.ThrowIfCancellationRequested();
 
-                bool reportDiagnosticsForCurrentTrivia = trivia.SyntaxTree.ReportDocumentationCommentDiagnostics();
-
-                if (!processedDocComment)
+                if (documentedParameters != null)
                 {
-                    // Since we have to throw away all the parts if any part is bad, we need to write to an intermediate temp.
-                    BeginTemporaryString();
+                    foreach (ParameterSymbol parameter in GetParameters(symbol))
+                    {
+                        if (!documentedParameters.Contains(parameter))
+                        {
+                            Location location = parameter.GetFirstLocation();
+                            Debug.Assert(location.SourceTree!.ReportDocumentationCommentDiagnostics()); //Should be the same tree as for the symbol.
 
-                    if (_processIncludes)
+                            // NOTE: parameter name, since the parameter would be displayed as just its type.
+                            _diagnostics.Add(ErrorCode.WRN_MissingParamTag, location, parameter.Name, symbol);
+                        }
+                    }
+                }
+
+                if (documentedTypeParameters != null)
+                {
+                    PooledHashSet<string> documentedTypeParameterNames = PooledHashSet<string>.GetInstance();
```

Comment:
>documentedTypeParameterNames [](http://example.com/codeflow?start=42&length=28)

Does it make sense to build this set from the start, or there are other consumers that should use the original set?

Reply:
Could you provide details why the switch to hashset of names was reverted?